js/store/reducers/onboarding.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * A reducer for the Onboarding.
+ * @flow
+ */
+
+import { TOS_ACCEPT_SUCCESS } from '../actions/constants'
+import { type Action } from '../../actions/types'
+import { type GlobalState } from '../../reducers/types'
+
+export type OnboardingState = {
```

Comment:
here we could restrict the allowed states (e.g. `isTosAccepted=false, isPinCreated=true` should not be a valid state

Reply:
```js
type OnboardingState = { isTosAccepted: false, isPinCreated: false } | { isTosAccepted: true, isPinCreated: false } | { isTosAccepted: true, isPinCreated: true };
```

Reply:
As i said in another comment the idea is:
* If ToS get updated, we put isTosAccepted = false and restart the onboarding (all other steps get skipped automatically) so in this case the state will be `{ isTosAccepted = false, isPinCreated: true }`
* If the user forget the PIN we use the same strategy, we set isPinCreated and restart the onboarding (all other steps get again skipped automatically)

The Onboarding state could be every variables combination.